src/engine/controls/keycontrol.cpp

```diff
@@ -24,8 +24,8 @@ KeyControl::KeyControl(const QString& group,
     m_pitchRateInfo.keylock = false;
 
     // pitch is the distance to the original pitch in semitones
-    // knob in semitones; 9.4 ct per midi step allowOutOfBounds = true;
-    m_pPitch = new ControlPotmeter(ConfigKey(group, "pitch"), -6.0, 6.0, true);
+    // knob in semitones; 9.4 ct per midi step allowOutOfBounds = false;
+    m_pPitch = new ControlPotmeter(ConfigKey(group, "pitch"), -30.0, 30.0, false);
```

Comment:
The limits here are for the range of the midi values mapped to the control. So this change might affect existing mappings. 

A better approach would be to use https://github.com/mixxxdj/mixxx/blob/eff27abd571ac3e1b273ad4aef6407eceee0cfee/src/mixer/basetrackplayer.cpp#L99
This allows to reject or setAndConfirm() CO updates.